common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -899,14 +899,17 @@ public RefreshTokenRecord getFamilyRefreshTokenForHomeAccountId(@NonNull final S
                                                            @NonNull AccountRecord accountRecord) {
         final List<IdTokenRecord> result = new ArrayList<>();
 
+        final List<Credential> allCredentials = mAccountCredentialCache.getCredentials();
```

Comment:
So, thing is, the only time that it should be decrypting these values is when there is more than 256 entries in the cache.  We could tune that value.

Reply:
ACK - I'll go ahead and merge this change in as-is. I would love to get take your input very soon on identifying and weighing bang-for-buck opportunities to squeeze even more performance out of this layer. Work I am happy to do and think it will benefit from more eyes / creative solutions